services/gateway/gateway-service.mjs

```diff
@@ -181,14 +187,13 @@ class GatewayService {
     // Synchronize the state.
     let firstLocation = {playerID: playerID, 
       displayName: socket.playerData.displayName};
-    firstLocation.x = socket.playerData.x;
-    firstLocation.y = socket.playerData.y;
+    firstLocation.mapCoord = socket.playerData.mapCoord;
     firstLocation.facing = 'D';
     firstLocation.displayChar = socket.playerData.displayChar;
     socket.playerData.lastMovingTime = Date.now();
-    //try occupy grid
-    await this.dir.redis.setAsync(
-      [`${firstLocation.x}-${firstLocation.y}`, 'occupied', 'NX']);
+    // try occupy grid
+    // TODO: Handle cases whereby we can't occupy the location.
+    await this._occupyCoord(firstLocation.mapCoord, playerID);
```

Comment:
I think for now we should use a while loop to guard this function call instead of let it go.
```
while (! await this._occupyCoord(firstLocation.mapCoord, playerID));
```
Imagine that there is no spawn point that is not occupied. We should not let the player spawn since it might mess something later.

Reply:
Lisa's PR that's coming up soon will resolve this, so I would prefer leaving this here.